src/core/types/chat.rs

```diff
@@ -1,18 +1,18 @@
 use crate::core::types::{ChatPermissions, ChatPhoto, Message};
 
-#[derive(Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[derive(Debug, Deserialize, Eq, Hash, PartialEq)]
 pub struct Chat {
     #[serde(rename = "chat_id")]
-    pub id: i32,
+    pub id: i64,
     #[serde(flatten)]
-    pub type_: ChatType,
+    pub chat_kind: ChatKind,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub photo: Option<ChatPhoto>,
 }
 
-struct PrivateChatTypeVisitor;
+struct PrivateChatKindVisitor;
```

Comment:
`PrivateChatKindVisitor`, `impl<'de> serde::de::Visitor<'de>` and `assert_private_field` should be moved behind the `ChatKind` and `NonPrivateChatKind`, I think.

Reply:
Okay